components/authz-service/server/v2/policy_refresher.go

```diff
@@ -51,8 +52,15 @@ func (m *policyRefresherMessageRefresh) Err() error {
 	return <-m.status
 }
 
-func NewPolicyRefresher(ctx context.Context, log logger.Logger, store storage.Storage,
-	engine engine.V2pXWriter) (PolicyRefresher, error) {
+func NewPostgresPolicyRefresher(ctx context.Context, log logger.Logger, engine engine.V2pXWriter) (PolicyRefresher, error) {
+	store := postgres.GetInstance()
+	if store == nil {
```

Comment:
When would this occur? Shouldn't it be waiting for it rather than just fail?

Reply:
This would happen if someone calls NewPostgresPolicyRefresher without initializing the postgres singleton as the error message suggests.